include/smack/SmackRep.h

```diff
@@ -150,6 +166,14 @@ class SmackRep {
   virtual const Stmt* alloca(llvm::AllocaInst& i);
   virtual const Stmt* memcpy(const llvm::MemCpyInst& msi);
   virtual const Stmt* memset(const llvm::MemSetInst& msi);
+  virtual const Stmt* loadAsBytes(const llvm::LoadInst& li);
+  virtual const Stmt* storeAsBytes(const llvm::StoreInst& si);
```

Comment:
I tried to implement `SmackRep::load` and `SmackRep::store` as you suggest. You could review the modification on the lasted commit. I still found it kind of hard to get `llvm::Function` from `addr` or `val` so that I add a default `llvm::Instruction` argument. 

For storing value into globals in `SmackRep::addInit`, it is inconsistent with the usage of `SmackRep::store` because it is impossible to get the pointer of type `llvm::Value` if it points to certain field of aggregate type globals. So I stick to the old implementation on this point. 

In sum, I think my implementation is still dirty. Your further suggestion is really appreciated.